packages/core/src/provider/types.ts

```diff
@@ -78,9 +81,28 @@ export type PaymasterAndDataMiddleware = AccountMiddlewareOverrideFn<
 export type GasEstimatorMiddleware = AccountMiddlewareOverrideFn<
   "callGasLimit" | "preVerificationGas" | "verificationGasLimit"
 >;
+export type GasEstimatorFeeOptions = Partial<
+  Pick<
+    UserOperationFeeOptions,
+    "callGasLimit" | "preVerificationGas" | "verificationGasLimit"
+  >
+>;
+
 export type FeeDataMiddleware = AccountMiddlewareOverrideFn<
   "maxFeePerGas" | "maxPriorityFeePerGas"
 >;
+export type FeeDataFeeOptions = Partial<
+  Pick<UserOperationFeeOptions, "maxFeePerGas" | "maxPriorityFeePerGas">
+>;
+
+export type FeeOptionsMiddleware = AccountMiddlewareOverrideFn<
```

Comment:
how is this middleware used? feels like it combines the fee estimator and gas estimator middlewares? Do we really need it? I'm concerned this becomes confusing for devs

Reply:
discussed offline